app/main.py

```diff
@@ -1,8 +1,24 @@
 class Person:
-    # write your code here
-    pass
+    people = dict()
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[self.name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list:
+    person_list = [
+        Person(person["name"], person["age"]) for person in people
+    ]
+
+    for person in people:
+        person_name = person["name"]
+        person_object = Person.people[person_name]
+
+        if person.get("wife"):
+            person_object.wife = Person.people[person["wife"]]
+        elif person.get("husband"):
+            person_object.husband = Person.people[person["husband"]]
```

Comment:
The code assumes that if a person has a 'wife' or 'husband', they are already present in the `Person.people` dictionary. This assumption might not hold true if the input data is incomplete or unordered. Consider adding checks to ensure the referenced person exists in the dictionary before accessing it.